src/components/tooltip/tooltip.js

```diff
@@ -99,12 +104,30 @@ function MdTooltipDirective($timeout, $window, $$rAF, $document, $mdUtil, $mdThe
         angular.element($window).off('resize', debouncedOnResize);
       });
 
-      scope.$watch('visible', function (isVisible) {
-        if (isVisible) showTooltip();
-        else hideTooltip();
-      });
+      if (element[0] && 'MutationObserver' in $window) {
```

Comment:
While $observe is technically faster than $watch since it still since it's a string comparison vs an object/expression evaluation, both are checked synchronously every $digest. 

MutationObserver is handled by the browser and signaled asynchronously.